boot-speed/clouds/ec2_cleanup.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+"""Clean up all running bootspeed-* EC2 instances.
+
+Copyright 2018-2019 Canonical Ltd.
+Joshua Powers <josh.powers@canonical.com>
+Paride Legovini <paride.legovini@canonical.com>
+"""
+import boto3
+import datetime as dt
+import re
+
+
+def clean_ec2():
+    """Clean up all running EC2 instances tagged 'bootspeed-*'."""
+    resource = boto3.resource('ec2')
+
+    inst_tag = [{'Name': 'tag:Name', 'Values': ['bootspeed-*']}]
+    for vpc in list(resource.vpcs.all()):
+        print('# cleaning up vpc %s\n' % vpc.id)
+        for instance in vpc.instances.filter(Filters=inst_tag):
+            for tag in instance.tags:
+                if tag['Key'] == 'Name' and re.match(
+                        "^bootspeed-[0-9]+-", tag['Value']):
+                    print('Found bootspeed instance %s' % instance.id)
+                    timestamp = int(tag['Value'].split("-")[1])
+                    tdelta = int(dt.datetime.utcnow().timestamp()) - timestamp
+                    print('Instance started %s seconds ago' % tdelta)
+                    if tdelta > 4*3600:
+                        print("Deleting stale instance.")
+                        instance.terminate()
+                        instance.wait_until_terminated()
```

Comment:
This script is likely to run much faster if you separate the initiate-termination and wait-for-termination in to separate phases (because then all the terminations are happening in parallel). (I think this should be as simple as appending `instance` to a list and then just iterating over that list calling `wait_until_terminated`?)